specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2018-06-01-preview/extensions.json

```diff
@@ -0,0 +1,357 @@
+                "x-ms-examples": {
+                    "Enable cluster monitoring": {
+                        "$ref": "./examples/GetLinuxClusterMonitoringStatus.json"
+                    }
+                },
+                "operationId": "Extensions_GetMonitoringStatus",
+                "parameters": [
+                    {
+                        "$ref": "#/parameters/SubscriptionIdParameter"
+                    },
+                    {
+                        "$ref": "#/parameters/ResourceGroupNameParameter"
+                    },
+                    {
+                        "$ref": "#/parameters/ClusterNameParameter"
+                    },
+                    {
+                        "$ref": "#/parameters/ApiVersionParameter"
+                    }
+                ],
+                "responses": {
+                    "200": {
+                        "description": "OK. The request has succeeded.",
+                        "schema": {
+                            "$ref": "#/definitions/ClusterMonitoringResponse"
+                        }
+                    }
+                }
+            },
+            "delete": {
+                "tags": [
+                    "Extensions"
+                ],
+                "description": "Disables the Operations Management Suite (OMS) on the HDInsight cluster.",
+                "x-ms-examples": {
+                    "Enable cluster monitoring": {
+                        "$ref": "./examples/DisableLinuxClusterMonitoring.json"
+                    }
+                },
+                "operationId": "Extensions_DisableMonitoring",
+                "parameters": [
+                    {
+                        "$ref": "#/parameters/SubscriptionIdParameter"
+                    },
+                    {
+                        "$ref": "#/parameters/ResourceGroupNameParameter"
+                    },
+                    {
+                        "$ref": "#/parameters/ClusterNameParameter"
+                    },
+                    {
+                        "$ref": "#/parameters/ApiVersionParameter"
+                    }
+                ],
+                "responses": {
+                    "200": {
+                        "description": "Ok response definition."
+                    },
+                    "202": {
+                        "description": "OK response definition."
+                    }
+                },
+                "x-ms-long-running-operation": true
+            }
+        },
+	
+        "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.HDInsight/clusters/{clusterName}/extensions/{extensionName}": {
+            "put": {
+                "tags": [
+                    "Extensions"
+                ],
+                "description": "Creates an HDInsight cluster extension.",
+                "operationId": "Extensions_Create",

```

Comment:
>"operationId": "Extensions_Create", [](start = 16, length = 35)

if the Update of the extension is allowed and it overwrites the existing one - then this operationId should be `Extensions_CreateOrUpdate`

Reply:
We don't allow updates on the extension.